### PR TITLE
kata-msft: support images with VOLUME directives

### PR DESCRIPTION
**Pull request by miampf**
Remove the `volumeMounts` fix since https://github.com/kata-containers/kata-containers/pull/10136 was merged

### CODE REVIEW

File: packages/by-name/microsoft/genpolicy/0007-genpolicy-support-for-VOLUME-definition-in-container.patch [hunk 1]
```diff
@@ -0,0 +1,641 @@
...
++
++    storages.push(agent::Storage {
++        driver: settings_image.driver.clone(),
++        driver_options: Vec::new(),
++        source: settings_image.source.clone(),
++        fstype: settings_image.fstype.clone(),
++        options: settings_image.options.clone(),
++        mount_point: destination_string.clone(),
++        fs_group: None,
++    });
```
Comment (burgerdev) [inline packages/by-name/microsoft/genpolicy/0007-genpolicy-support-for-VOLUME-definition-in-container.patch:203]: Looks like this may be wrong upstream.

Comment (miampf) [reply packages/by-name/microsoft/genpolicy/0007-genpolicy-support-for-VOLUME-definition-in-container.patch]: Removed it, stuff works now :)

File: packages/by-name/microsoft/genpolicy/genpolicy_msft_settings_coordinator.patch [hunk 2]
```diff
@@ -12,10 +12,6 @@ index 7d35862a..4eacc7cd 100644
              ]
          },
          "CopyFileRequest": [
-@@ -322,4 +323,4 @@
-         "UpdateEphemeralMountsRequest": false,
-         "WriteStreamRequest": false
-     }
```
Comment (burgerdev) [inline packages/by-name/microsoft/genpolicy/genpolicy_msft_settings_coordinator.patch:18]: Why is this necessary?

File: packages/by-name/microsoft/genpolicy/genpolicy_msft_settings_dev.patch [hunk 3]
```diff
@@ -27,6 +27,7 @@ index 7d35862a..f469b201 100644
 -        "WriteStreamRequest": false
 +        "WriteStreamRequest": true
      }
--}
-\ No newline at end of file
-+}
+ }
+-- 
+2.47.0
+
```
Comment (burgerdev) [inline packages/by-name/microsoft/genpolicy/genpolicy_msft_settings_dev.patch:33]: Please make sure not to include git versions here.

File: packages/by-name/microsoft/genpolicy/package.nix [hunk 4]
```diff
@@ -55,6 +55,10 @@ rustPlatform.buildRustPackage rec {
       # We can revisit this if microsoft upstreamed
       # https://github.com/microsoft/kata-containers/pull/174
       ./0006-genpolicy-support-HostToContainer-mount-propagation.patch
+      # This patch is a port of https://github.com/kata-containers/kata-containers/pull/10136/files
+      # to microsofts genpolicy.
+      # remove when picked up by Microsoft/kata-containers fork.
+      ./0007-genpolicy-support-for-VOLUME-definition-in-container.patch
```
Comment (burgerdev) [inline packages/by-name/microsoft/genpolicy/package.nix:61]:
```suggestion
      # This patch is a port of https://github.com/kata-containers/kata-containers/pull/10136
      # to Microsoft's genpolicy.
      # TODO(miampf): remove when picked up by microsoft/kata-containers fork.
      ./0007-genpolicy-support-for-VOLUME-definition-in-container.patch
```

File: packages/by-name/microsoft/genpolicy/0007-genpolicy-support-for-VOLUME-definition-in-container.patch [hunk 5]
```diff
@@ -0,0 +1,631 @@
...
++                policy_mounts,
++                storages,
++                volume.0,
++            );
++        }
++    }
+ }
+ 
+ /// Add the "io.katacontainers.config.agent.policy" annotation into
+diff --git a/tests/integration/kubernetes/k8s-policy-deployments.bats b/tests/integration/kubernetes/k8s-policy-deployments.bats
```
Comment (burgerdev) [inline packages/by-name/microsoft/genpolicy/0007-genpolicy-support-for-VOLUME-definition-in-container.patch:525]: Consider removing the test scripts from the patch - we're not going to use them.

**Review by burgerdev — COMMENTED** [review]
Great, thanks!

File: packages/by-name/microsoft/genpolicy/0001-genpolicy-add-rules-and-types-for-volumeDevices.patch [hunk 6]
```diff
@@ -1,7 +1,7 @@
-From 41f26a5803fa50abf3bd0d6cfebc8106ae9dcbc8 Mon Sep 17 00:00:00 2001
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: Markus Rudy <mr@edgeless.systems>
 Date: Thu, 23 May 2024 09:20:20 +0200
-Subject: [PATCH 1/6] genpolicy: add rules and types for volumeDevices
+Subject: [PATCH 1/7] genpolicy: add rules and types for volumeDevices
```
[the remark this answers is not in the document]
Comment (miampf) [reply packages/by-name/microsoft/genpolicy/0001-genpolicy-add-rules-and-types-for-volumeDevices.patch:4]: I did generate them using `-N` and then manually rewrote them since the patch set on main used the `PATCH n/m` formatting. Should I just remove that so only `[PATCH]` is written everywhere?

File: packages/by-name/microsoft/genpolicy/genpolicy_msft_settings_coordinator.patch (hunk 2 above)
Comment (miampf) [reply packages/by-name/microsoft/genpolicy/genpolicy_msft_settings_coordinator.patch:18]: I think that was an accident, will revert that part.

Comment (miampf) [reply packages/by-name/microsoft/genpolicy/genpolicy_msft_settings_coordinator.patch:18]: Nvm, I did some digging, I removed that part because it is added by default in the new patch. I didn't make the settings changes into a "settings patch" since, following the PR in msft.kata-containers, they have to be there for all settings instead of being some specialization.

File: packages/by-name/microsoft/genpolicy/0001-genpolicy-add-rules-and-types-for-volumeDevices.patch (hunk 6 above)
[the remark this answers is not in the document]
Comment (burgerdev) [reply packages/by-name/microsoft/genpolicy/0001-genpolicy-add-rules-and-types-for-volumeDevices.patch]: Yes, not all patch sets are converted to the new standard. Please just use `git format-patch` as documented.

**Review by burgerdev — APPROVED** [review]

**Comment by burgerdev** [comment]
https://github.com/edgelesssys/contrast/actions/runs/12028677575/job/33539834180?pr=996#step:11:273 is a Nydus flake, so feel free to merge after fixing the commit structure (or squash).
